lib/honeybadger/insights/ecto.ex

```diff
@@ -0,0 +1,75 @@
+defmodule Honeybadger.Insights.Ecto do
+  @moduledoc false
+
+  use Honeybadger.Insights.Base
+
+  @required_dependencies [Ecto.Repo]
+  @telemetry_events []
+
+  @excluded_queries [
+    ~r/^(begin|commit)( immediate)?( transaction)?$/i,
+    # Also exclude pg_notify which is often used with Oban
+    ~r/SELECT pg_notify/,
+    ~r/schema_migrations/
+  ]
+
+  @excluded_sources [
+    "schema_migrations",
+    "oban_jobs",
+    "oban_peers"
+  ]
+
+  def get_telemetry_events do
+    :ecto_repos
+    |> Honeybadger.get_env()
+    |> Enum.map(&get_telemetry_prefix/1)
+  end
+
+  def extract_metadata(meta, _name) do
+    meta
+    |> Map.take([:query, :decode_time, :query_time, :queue_time, :source])
+    |> Map.update!(:query, &obfuscate(&1, meta.repo.__adapter__()))
+  end
+
+  def ignore?(%{query: query, source: source}) do
+    if source in get_insights_config(:excluded_sources, @excluded_sources) do
+      true
+    else
+      :excluded_queries
+      |> get_insights_config(@excluded_queries)
+      |> Enum.any?(fn
+        pattern when is_binary(pattern) -> query == pattern
+        %Regex{} = pattern -> Regex.match?(pattern, query)
+        _pattern -> false
+      end)
+    end
+  end
+
+  defp get_telemetry_prefix(repo) do
+    Keyword.get(repo.config(), :telemetry_prefix, []) ++ [:query]
+  end
+
+  @escape_quotes ~r/(\\\"|\\')/
+  @squote_data ~r/'(?:[^']|'')*'/
+  @dquote_data ~r/"(?:[^"]|"")*"/
+  @number_data ~r/\b\d+\b/
+  @double_quoters ~r/(postgres|sqlite3|postgis)/i
```

Comment:
I think it's more idiomatic to keep these kinds of variables at the top of the module, but they are only used in the function below so I thought it made sense to keep them close.